### ISSUE TITLE
snapToItem(0) doesn't work

### ISSUE THREAD

**Issue by arnaud-cortisse** [archriss/react-native-snap-carousel#685]
### Is this a bug report, a feature request, or a question?

Bug report

### Have you followed the required steps before opening a bug report?

- [x] I have read [the guidelines regarding bug report](https://github.com/archriss/react-native-snap-carousel/blob/master/CONTRIBUTING.md).
- [x] I have reviewed [the documentation](https://github.com/archriss/react-native-snap-carousel/blob/master/README.md) in its entirety, including the dedicated documentations :books:.
- [x] I have searched for [existing issues](https://github.com/archriss/react-native-snap-carousel/issues) and made sure that the problem hasn't already been reported.
- [] I am using [the latest plugin version](https://github.com/archriss/react-native-snap-carousel/releases). --> I'm using ^4.0.0-beta.2
- [x] I am following [the issue template](https://raw.githubusercontent.com/archriss/react-native-snap-carousel/master/ISSUE_TEMPLATE.md) closely in order to produce a useful bug report.

### Have you made sure that it wasn't a [React Native bug](https://github.com/archriss/react-native-snap-carousel/issues?utf8=%E2%9C%93&q=is%3Aissue+label%3A%22react-native+bug%22+)?

Yes

### Is the bug specific to iOS or Android? Or can it be reproduced on both platforms?

It's on both. 

### Is the bug reproductible in a production environment (not a debug one)?

Yes.

### Environment

  Environment:
    React: 16.9.0
    React native: 0.61.
    react-native-snap-carousel: ^4.0.0-beta.2

  Target Platform:
    Android (10.0)

### Expected Behavior

snapToItem(0) works properly.

### Actual Behavior

snapToItem(0) doesn't work. The carousel doesn't move at all. 
It seems to be working with any other index.

### Reproducible Demo

I can't get the example to work with the beta version in the snack, but I can get it to work locally with a new expo project (it complains about the beta dependency).
https://snack.expo.io/@acortisse/react-native-snap-carousel-playground

### Steps to Reproduce

1. Press "Snap to index 1"
2. Press "Snap to index 2"
3. Press "Snap to index 0"

**Comment by husenLogicwind** [comment]
i'm facing same issue.

**Comment by tleuschner** [comment]
In the `_snapToItem` method there seems to be an issue handling the offset value 0. To be precise this piece of code in the method will always return if the offset of the first item is 0.

     
```
const offset = this._getItemScrollOffset(index);

if (!offset) {
    return;
}
```

I simply changed checking the offset value to 

```
if (offset < 0) {
    return;
}
```

and it seems to be working fine. As I do not fully know the Carousel, I can't say that this won't bring up any other issues.

**Comment by bd-arc** [comment]
Fixed in `4.0.0-beta.4`.
